### PR TITLE
1197 admin interface created transactions list

### LINKED ISSUE #1197: 🚀 [Feature] Admin interface created transactions list
## 🚀 Feature
it should be visible for the user search / user individual view in the admin area:
- [ ] List of created transactions
- [ ] List of all transactions without text for this user

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
in the user search and user overview in the admin area, there is a button for each user to list all creations of gdd for the user. 

### Issues
- fixes  #1197 

### merged
- #1201 
- #1199 
 
FireShot Capture 866 - Gradido Admin Interface - localhost

FireShot Capture 867 - Gradido Admin Interface - localhost

### CODE REVIEW

File: admin/src/components/CreationTransactionListFormular.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        {
+          gdd: 380,
+          date: '22.04.2021',
+          memo: 'bnnbnbbncvbcv bcbcbc bcb cbc bcb cvb cbccvvbcvb bvbcbcbcbc bcb cb',
+          moderator: 'Moderator Zwei',
+        },
+      ],
+    }
+  },
+}
```
Comment (ulfgebhardt) [inline admin/src/components/CreationTransactionListFormular.vue:58]: We need real data here

File: admin/src/components/ConfirmRegisterMailFormular.vue [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        .then(() => {
+          // eslint-disable-next-line no-console
+          console.log(
+            'Erfolgreich senden der Confirmation Link an die E-Mail des Users!',
+            this.email,
+          )
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.log('Fehler beim senden des confirmation link an den Benutzer', error)
```
Comment (ulfgebhardt) [inline admin/src/components/ConfirmRegisterMailFormular.vue:53]: toast this error

File: admin/src/components/UserTable.vue [hunk 3]
```diff
@@ -181,6 +233,64 @@ export default {
...
+      if (details) {
+        row.toggleDetails()
+      }
+      if (!details) {
+        row.toggleDetails()
+        this.showCreationFormular = true
+        if (this.$refs.showing_detals_true !== undefined) {
+          this.$refs.showing_detals_true.click()
+        }
+      }
```
Comment (ulfgebhardt) [inline admin/src/components/UserTable.vue:255]: this had issues when switching between two users
1. Open a dialog X for User A
2. Open a dialog Y for User B
-> Problem?

File: admin/src/pages/UserSearch.vue [hunk 4]
```diff
@@ -48,6 +57,11 @@ export default {
   },
 
   methods: {
+    unconfirmedRegisterMails() {
+      this.searchResult = this.searchResult.filter((result) => {
```
Comment (Mogge) [inline admin/src/pages/UserSearch.vue:61]: Please do this in the backend. When we have pagination then everything breaks with this solution

**Review by Mogge — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/TransactionResolver.ts [hunk 5]
```diff
@@ -379,6 +380,11 @@ async function listTransactions(
 
   const transactionList = new TransactionList()
   transactionList.count = userTransactionsCount
+  if (onlyCreations) {
+    transactions = transactions.filter((transaction) => {
+      return transaction.type === 'creation'
```
Comment (Mogge) [inline backend/src/graphql/resolver/TransactionResolver.ts:385]: This way we will have problems with the pagination. Can we filter in the query?

File: backend/src/graphql/resolver/AdminResolver.ts [hunk 6]
```diff
@@ -315,3 +318,10 @@ function isCreationValid(creations: number[], amount: number, creationDate: Date
   }
   return true
 }
+async function hasActivatedEmail(email: string): Promise<boolean> {
+  const repository = getCustomRepository(LoginUserRepository)
+  const user = await repository.findByEmail(email)
+  let emailActivate = false
+  if (user) emailActivate = user.emailChecked
+  return emailActivate
```
Comment (Mogge) [inline backend/src/graphql/resolver/AdminResolver.ts:326]:
```suggestion
  return user ? user.emailChecked : false
```
And no variable needed

File: admin/src/components/CreationTransactionListFormular.vue [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            onlyCreations: true,
+            userId: parseInt(this.userId),
+          },
+        })
+        .then((result) => {
+          this.items = result.data.transactionList.transactions
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.log('Something went wrong', error)
```
Comment (Mogge) [inline admin/src/components/CreationTransactionListFormular.vue:37]: Toast the error please /and test it)

**Review by Mogge — CHANGES_REQUESTED** [review]

File: admin/src/components/ConfirmRegisterMailFormular.vue (hunk 2 above)
Comment (Elweyn) [reply admin/src/components/ConfirmRegisterMailFormular.vue:53]: Replaced in then with this.$toasted.global.success, and in catch this.$toasted.global.error

File: admin/src/components/CreationFormular.vue [hunk 8]
```diff
@@ -1,5 +1,6 @@
 <template>
   <div class="component-creation-formular">
+    CREATION FORMULAR
```
Comment (Elweyn) [inline admin/src/components/CreationFormular.vue:3]: Why do we have added this Text hier?

File: admin/src/components/CreationTransactionListFormular.vue (hunk 7 above)
Comment (Elweyn) [reply admin/src/components/CreationTransactionListFormular.vue]: Toast has been added

File: backend/src/config/index.ts [hunk 9]
```diff
@@ -52,6 +52,7 @@ const email = {
   EMAIL_LINK_VERIFICATION:
     process.env.EMAIL_LINK_VERIFICATION || 'http://localhost/vue/checkEmail/$1',
   EMAIL_LINK_SETPASSWORD: process.env.EMAIL_LINK_SETPASSWORD || 'http://localhost/vue/reset/$1',
+  RESEND_TIME: process.env.RESEND_TIME || 10,
```
Comment (Elweyn) [inline backend/src/config/index.ts:55]: Maybe check if process.env.RESEND_TIME isNum

**Review by Elweyn — CHANGES_REQUESTED** [review]

File: backend/src/config/index.ts (hunk 9 above)
Comment (Mogge) [inline backend/src/config/index.ts:55]:
```suggestion
  RESEND_TIME: isNaN(process.env.RESEND_TIME) ? 10 : parseInt(process.env.RESEND_TIME),
```

Comment (Elweyn) [reply backend/src/config/index.ts:55]:
```suggestion
RESEND_TIME: process.env.RESEND_TIME
    ? isNaN(parseInt(process.env.RESEND_TIME))
      ? 10
      : parseInt(process.env.RESEND_TIME)
    : 10,
```

**Review by Mogge — APPROVED** [review]

**Review by Elweyn — APPROVED** [review]

**Pull request merged**